### PR TITLE
Upgrading to rails 5

### PR DESCRIPTION
**Pull request by KarimChukfeh**
Handled many deprecation warnings mostly related to the style of passing params in controller tests.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,16 +1,21 @@
 source 'https://rubygems.org'
 ruby '2.3.1'
 
-gem 'rails', '~> 4.2'
-gem 'thin', '~> 1.6.3'
+gem 'rails', '~> 5'
```
Comment (tahnok) [inline Gemfile]: Can you include at least one minor revision here? eg: 5.0 in this case

File: bin/update [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
```
Comment (tahnok) [inline bin/update]: What is this file for? I can't tell from reading it

File: config/cable.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+development:
```
Comment (tahnok) [inline config/cable.yml]: I don't think we want this? I believe it's from ActionCable and it's wrong in production

File: config/environments/development.rb [hunk 4]
```diff
@@ -27,50 +42,13 @@
...
-  config.assets.raise_runtime_errors = true
+  # Suppress logger output for asset requests.
+  config.assets.quiet = true
 
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
 
-  config.console = Pry
-
-  Octokit.configure do |c|
```
Comment (tahnok) [inline config/environments/development.rb]: Why were the following sections deleted?

File: config/puma.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
+# Puma can serve each request in a thread from an internal thread pool.
+# The `threads` method setting takes two numbers a minimum and maximum.
+# Any libraries that use thread pools should be configured to match
+# the maximum value specified for Puma. Default is set to 5 threads for minimum
+# and maximum, this matches the default thread size of Active Record.
+#
+threads_count = ENV.fetch("RAILS_MAX_THREADS") { 5 }.to_i
```
Comment (tahnok) [inline config/puma.rb]: Where did this come from? I see that we have included `thin` which I believe is an alternative to `puma` for the webserver?

File: bin/update (hunk 2 above)
Comment (KarimChukfeh) [inline bin/update]: Many of these files were generated after running `rake rails:update`.  It was interactive and I was able to say `n` perhaps I should've done that?

[see 1.4 in the guide to upgrading Rails](http://edgeguides.rubyonrails.org/upgrading_ruby_on_rails.html#the-update-task)

File: Gemfile (hunk 1 above)
Comment (KarimChukfeh) [inline Gemfile]: done

File: config/cable.yml (hunk 3 above)
Comment (KarimChukfeh) [inline config/cable.yml]: I don't fully understand what's wrong. Should I delete the file or just the first 3 lines?

File: config/environments/development.rb (hunk 4 above)
Comment (KarimChukfeh) [inline config/environments/development.rb]: automatic file generation. adding it back

File: config/puma.rb (hunk 5 above)
Comment (KarimChukfeh) [inline config/puma.rb]: deleted

File: config/cable.yml (hunk 3 above)
Comment (KarimChukfeh) [inline config/cable.yml]: deleted

Comment (tahnok) [inline config/cable.yml]: Should probably still delete this if we're not using ActionCable

File: config/puma.rb (hunk 5 above)
Comment (tahnok) [inline config/puma.rb]: Seems like it's still here?

Comment (KarimChukfeh) [inline config/puma.rb]: woops 😅

File: config/cable.yml (hunk 3 above)
Comment (KarimChukfeh) [inline config/cable.yml]: I don't think it's there anymore

**Comment by tahnok** [comment]
I think this branch can wait for now. There are more important things to do before september comes. Maybe a few weeks into the semester we can return to this?

**Comment by KarimChukfeh** [comment]
check rails_5_branch instead

**Pull request closed without merging**